pkg/receiver/activedirectoryinvreceiver/factory.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package activedirectoryinvreceiver
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "activedirectoryinv"
+)
+
+// NewFactory creates a factory for Active Directory Inventory receiver
+func NewFactory() receiver.Factory {
+	return receiver.NewFactory(
+		typeStr,
+		CreateDefaultConfig,
+		receiver.WithLogs(createLogsReceiver, component.StabilityLevelAlpha),
+	)
+}
+
+// CreateDefaultConfig creates the default configuration for the receiver
+func CreateDefaultConfig() component.Config {
+	return &ADConfig{
+		DN:           "CN=Guest,CN=Users,DC=exampledomain,DC=com",
```

Comment:
Yeah, I think this should be an empty string by default, since it comes under optional arguments in the [docs](https://help.sumologic.com/docs/send-data/installed-collectors/sources/windows-active-directory-inventory-source/)

Reply:
Yes let's make it empty by default if it's optional.

Reply:
I think I'm getting confused on this one. If an empty `base_dn` makes the receiver to not function property, then the empty value should probably be forbidden, making `base_dn` a required property. Is this correct?

Reply:
It wouldn't collect any data. The user would get an error message saying that the DN is invalid and the collector would keep polling according to the poll interval.

Reply:
As per our discussion, won't let the collector start without a valid non empty DN.